### ISSUE TITLE
Be resilient to "authority type not supported" errors

### ISSUE THREAD

**Issue by codingllama** [gravitational/teleport#39809]
__Expected behavior:__

Unknown authority types may appear during version downgrades. As far as reasonable, Teleport should both support downgrades and be resilient to unknown authority types.

__Current behavior:__

Example stack trace from a v13.4.17 downgraded to v13.4.14:

```
Original Error: *trace.BadParameterError &#34;db_client&#34; authority type is not supported
Stack Trace:
    github.com/gravitational/teleport/api@v0.0.0/types/trust.go:87 github.com/gravitational/teleport/api/types.CertAuthType.Check
    github.com/gravitational/teleport/api@v0.0.0/types/trust.go:102 github.com/gravitational/teleport/api/types.(*CertAuthID).Check
    github.com/gravitational/teleport/api@v0.0.0/types/authority.go:332 github.com/gravitational/teleport/api/types.(*CertAuthorityV2).CheckAndSetDefaults
    github.com/gravitational/teleport/lib/services/authority.go:64 github.com/gravitational/teleport/lib/services.ValidateCertAuthority
    github.com/gravitational/teleport/lib/services/authority.go:470 github.com/gravitational/teleport/lib/services.UnmarshalCertAuthority
    github.com/gravitational/teleport/lib/services/local/trust.go:278 github.com/gravitational/teleport/lib/services/local.(*CA).GetCertAuthorities
    github.com/gravitational/teleport/lib/cache/collections.go:955 github.com/gravitational/teleport/lib/cache.certAuthorityExecutor.getAll
    github.com/gravitational/teleport/lib/cache/collections.go:95 github.com/gravitational/teleport/lib/cache.(*genericCollection[...]).fetch
    github.com/gravitational/teleport/lib/cache/cache.go:1436 github.com/gravitational/teleport/lib/cache.(*Cache).fetch.func2
    golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 golang.org/x/sync/errgroup.(*Group).Go.func1
    runtime/asm_arm64.s:1172 runtime.goexit
User Message: failed to fetch resource: &#34;cert_authority&#34;
    &#34;db_client&#34; authority type is not supported] cache/cache.go:965
```

__Bug details:__
- Teleport version: all current
- Recreation steps: see above
- Debug logs: see above

**Comment by codingllama** [comment]
Assigned to Gavin in light of this being directly related to #35951, although we probably could find other possible offenders if we looked for them.

**Comment by GavinFrazar** [comment]
This is unfortunately caused by a bug in our range functions that I fixed in https://github.com/gravitational/teleport/pull/36148

That fix made it into `13.4.15`. Downgrading from 13.4.17 (ca split introducing new CA type) to 13.4.15 should work.

**Comment by GavinFrazar** [comment]
I'll test this to be sure though. If .17 -> .15 works then I'll close this ticket, as there's nothing else that I can do unfortunately.

**Comment by codingllama** [comment]
Thanks. If there are no follow ups feel free to close the issue.

**Comment by codingllama** [comment]
> I'll test this to be sure though. If .17 -> .15 works then I'll close this ticket, as there's nothing else that I can do unfortunately.

Sounds good!

**Comment by GavinFrazar** [comment]
Also, for future reference: I also merged that fix in `v12.4.33`, so downgrading 13->12 should target v12.4.33+

**Comment by GavinFrazar** [comment]
downgrading from `13.4.17` to `13.4.15` works and does not display the error, so it's caused by the range function bug fixed that was fixed in 13.4.15.

**Comment by codingllama** [comment]
Thanks for double-checking all of this, Gavin.

**Comment by rahulinux** [comment]
Encountered similar issue when downgraded from `v14.3.20` ->  `v13.3.8`

**Comment by codingllama** [comment]
@rahulinux you are just too far behind in releases. According to the comments above we a landed a fix in v13.4.15, by definition a tag older than that can't have the fix.
